bolt/runtime/common.h

```diff
@@ -407,13 +431,19 @@ class Mutex {
 /// RAII wrapper for Mutex
 class Lock {
   Mutex &M;
+  uint64_t SignalMask[1] = {};
 
 public:
   Lock(Mutex &M) : M(M) {
+    __sigprocmask(SIG_BLOCK, MaskAllSignals, SignalMask);
     while (!M.acquire()) {
     }
   }
-  ~Lock() { M.release(); }
+
+  ~Lock() {
+      M.release();
+      __sigprocmask(SIG_SETMASK, SignalMask, (void *)0);
```

Comment:
Well, this is a C++ file, so I guess we can just use:
`  __sigprocmask(SIG_SETMASK, SignalMask, nullptr);`
instead of going with the C dialect of casting NULL or 0.

Reply:
Sure, I don't mind :)